dev/table_name_collector.h

```diff
@@ -33,16 +33,12 @@ namespace sqlite_orm {
 
             template<class F, class O>
             void operator()(F O::*, std::string alias = {}) const {
-                if(this->find_table_name) {
-                    table_names.insert(std::make_pair(this->find_table_name(typeid(O)), move(alias)));
-                }
+                table_names.emplace(this->find_table_name(typeid(O)), move(alias));
```

Comment:
best option is throwing an exception if `!this->find_table_name` cause if it is `null` then we get UB. We must prefer exceptions instead of UB

Reply:
No, calling an "empty" function is not UB, the function wrapper will throw an exception.
Otherwise there's no need checking `this->find_table_name == nullptr` all the time. It's an internal class anyway, and should be used correctly in the first place. An assertion would be more helpful than a repetitive runtime check that only grows your code size.
The only place where `table_name_collector` gets ever used directly (circumventing the collection callbacks) is in `serialize_get_all_impl`, which then also performs an AST iteration that goes into the void.

Reply:
ok I did not know about `[bad_function_call](https://www.cplusplus.com/bad_function_call)` exception. It is ok. Thanks